node/src/actors/chain_manager/mod.rs

```diff
@@ -251,6 +253,8 @@ pub struct BlockCandidate {
     pub reputation: Reputation,
     /// Vrf proof
     pub vrf_proof: Hash,
+    /// Prority info
+    pub priorities: Priorities,
```

Comment:
On the other hand, during validation we're calculating the `fee` and `weight` for each transaction in every candidate anyway. If we don't store the priorities here, we will need to repeat the same computation for the best candidate upon consolidation.

So it felt to me like this was one of these situations in which you can trade-off a ridiculous amount of memory for the good of saving some CPU time. We are talking about potentially dozens of calls to `transaction.body.weight()` and iterating over hundreds of inputs and outputs to add and subtract their values. Not to mention that this also implies resolving an output pointer for every input you find there...

In contrast, please take into account that keeping those `Priorities` structs in memory merely takes 34 bytes per candidate, and that deriving `Priorities` from the `fee` and `weight` for all block candidates takes only a multiplication and a division each :eyes:

Reply:
The problem is calculating the fee because of the input UTXOs, calculating the weight is almost free. And I'm not worried about the memory overhead but about the CPU time used to calculate the priorities in advance. The more block candidates then the more time spent validating them, so there must be a number of candidates where it is faster to calculate the priorities later and only once, when we know the best candidate. Maybe that number is 10, maybe 1000, who knows.

Reply:
Gotcha. Taking into account that deriving the priority is only one multiplication and one division, would you strongly oppose to leaving it as is?

Reply:
I thought that calculating the priority needs to iterate over all the transactions of the block?

Reply:
Not at all, because at the time we're validating, we already have the added up fee and weight, and this PR simply adds a call to `Priority::from_fee_weight`, which internally only does the division and the multiplication.

Reply:
Ah, I see. Then it shouldn't be a problem to leave it like that.